pandas/core/window/rolling.py

```diff
@@ -365,7 +365,10 @@ def _prep_values(self, values: ArrayLike) -> np.ndarray:
                 if isinstance(values, ExtensionArray):
                     values = values.to_numpy(np.float64, na_value=np.nan)
                 else:
-                    values = ensure_float64(values)
+                    if len(values) != 0 and np.iscomplex(values).all():
```

Comment:
Why `len(values) != 0`? You can just use `is_complex_dtype`

Reply:
My mistake. Have updated